herbert_core/classes/MPIFramework/@MessagesCppMPI/private/init_framework_.m

```diff
@@ -31,21 +41,25 @@
 try
     ver = cpp_communicator();
     if ~strncmpi(ver,'$Revision::',11)
-        error('MPI_MESSAGES:runtime_error',... 
-            'Can not probe C++ MPI communicator, returned version %s',...
+        error('MPI_MESSAGES:runtime_error',...
+            'Can not probe C++ MPI communicator, returned version type: %s is not recognized',...
             ver);
     end
 catch Err
     error('MPI_MESSAGES:runtime_error',...
         'Can not initialize MPI communicator, err message: %s',...
-        Err.message);    
+        Err.message);
 end
-
+obj.data_message_tag_ = MESS_NAMES.mess_id('data');
 if ~isempty(obj.mpi_framework_holder_)
     cpp_communicator('finalize',obj.mpi_framework_holder_);
 end
-[obj.mpi_framework_holder_,obj.task_id_,obj.numLabs_]= cpp_communicator('init');
-
-
-
-
+if test_mode
+    [obj.mpi_framework_holder_,obj.task_id_,obj.numLabs_]= ...
+        cpp_communicator('init_test_mode'...
+        ,obj.assync_messages_queue_length_,obj.data_message_tag_);
```

Comment:
comma placement

Reply:
fixed in the next ticket. Let's lieve it alone here to avoid merge conflicts.